File: AggressiveCows/cows.cpp
#include<bits/stdc++.h>
using namespace std;
 
int minDistance(vector<int> v, int C)
{
  std::vector<int> d;
  for (int i = 0; i < v.size()-1; i++)
  {
    d.push_back(v[i+1]-v[i]);
  }
 
  int l = *std::min_element(d.begin(), d.end());
  int h = std::accumulate(d.begin(), d.end(), 0);
 
  while (l < h) 
  {
    int allowedNo = 1;
    int x = (h+l)/2;
    int distance = 0;
    for (int i=0; i<d.size(); i++)
    {
      if (distance + d[i] <= x)
        distance = distance + d[i];
      else 
      {
        allowedNo++;
        distance = 0;
      }
    }
    if (allowedNo >= C)
      l = x + 1;
    else
      h = x;
  }
  return l;
}
 
int main() 
{
  int t, N, C, input, output;
  std::vector<int> o;
  cin>>t;
  while(t--)
  {
    cin>>N>>C;
    std::vector<int> v;
    while(N--)
    {
      cin>>input;
      v.push_back(input);
    }
    sort(v.begin(), v.end());
    output = minDistance(v, C);
    o.push_back(output);
  }
  for (int i=0; i<o.size(); i++) 
  {
    cout<<o[i]<<"\n";
  }
}

File: AggressiveCows/cows.py
import math

def minDistance(x, C):
    d = []
    for i in range(0, len(x)-1):
        d.append(x[i+1]-x[i])
        i = i+1

    l = min(d)
    h = sum(d)

    while l < h:
        allowedNo=1
        x = math.floor((h+l)/2)
        distance = 0
        for i in range(len(d)):
            if distance + d[i] <= x:
                distance = distance + d[i]
            else:
                allowedNo = allowedNo + 1
                distance = 0
            i = i+1
    
        if allowedNo >= C:
            l = x + 1
        else:
            h = x
    return l


x = []
numbers = []
t = int(input())
for i in range(t):
    NC = input()
    N, C = NC.split(" ")
    N = int(N)
    C = int(C)
    for j in range(N):
        x.append(int(input()))
    x.sort()
    numbers.append(minDistance(x, C))
for j in numbers:
    print(j)

